include/glow/Base/TensorSerialization.h

```diff
@@ -52,6 +54,32 @@ void dumpToRawTextFile(Tensor &tensor, llvm::StringRef filename);
 /// round-trips (dump/load) but is used for human readability.
 void loadFromRawTextFile(Tensor &tensor, llvm::StringRef filename);
 
+/// Load network input tensor (same one images are loaded into) from the tensor
+/// blobs files in \p filenames. All the loaded tensors are concatanated along
+/// the batch dimension. The default loader expected the following tensor blob
+/// format:
+///   -- first line: 4 space separated values representing NCHW dimensions (W is
+///   the fastest).
+///   -- second line: Space separated float value that are loaded into \p
+///   tensor.
+/// TODO: Default tensor laoder could be extended to support various data types
+/// or image layout.
+void loadInputTensorFromFileWithType(
```

Comment:
Since this method is specific to images can we name it more appropriately?

Reply:
Actually I don't see it as image related - it's simply is loading a tensor just like ModelProfiler does or TensorSerializer existing functions.
We still didn't have the need to load multiple inputs (like ModelProfiler does) so I've left this out.  Also, could merge with existing TensorSerializer functions.

Reply:
Well the reason I said this is because you pass in `ImageLayout`. Although it doesn't do anything if not set to the default, it would feel weird to use this with non-image Tensors.

And yeah probably makes sense to merge with TensorSerialization

Reply:
I've renamed imageLayout into tensorLayout. Layout can be specified for a tensor and the function also can be used to load images converted to tensors (so we don't have to provide various image format loaders).

Reply:
I would still prefer this to be called something like `loadInputImageFromFileWithType()` -- even if `imageLayout` is named `tensorLayout` it doesn't change that this API seems image-specific. E.g. the input must have a specific number of dimensions as it expects to load images. What are your concerns with renaming it? Do you actually use this with non-images?

Reply:
@jfix71 Take for example a stereo image where now input has 6 channels. For us, that's not image anymore - it's just some tensor, for all we care.
Sure the input has to be 4D, but for me that's just the limitation of this tensor loader, as we don't need anything better now. If that's a reason to call this image-specific, please confirm, I'll rename.

Reply:
@jfix71 , fixed anyway.